MyBlog.Services/Areas/Articles/ArticleService.cs

```diff
@@ -0,0 +1,154 @@
+using AutoMapper;
+using Microsoft.EntityFrameworkCore;
+using MyBlog.Infrastructure;
+using MyBlog.Infrastructure.Entities;
+using MyBlog.Infrastructure.Entities.Enum;
+using MyBlog.Service.Areas.Articles.AutoMapper.Dto;
+using MyBlog.Service.Exception;
+using MyBlog.Service.Helpers.TokenParser.Dto;
+
+namespace MyBlog.Service.Areas.Articles;
+
+public class ArticleService : IArticleService
+{
+    private readonly MyBlogContext _context;
+    private readonly IMapper _mapper;
+
+    public ArticleService(IMapper mapper, MyBlogContext context)
+    {
+        _mapper = mapper;
+        _context = context;
+    }
+
+    public async Task<List<ArticleDto>> GetListAsync()
+    {
+        var articles = await _context.Articles
+            .Select(article => _mapper.Map<ArticleDto>(article))
+            .ToListAsync();
+
+        ThrowIfEmptyList(articles);
+        
+        return articles;
+    }
+    
+    public async Task<List<ArticleDto>> GetByUserIdAsync(int userId)
+    {
+        var articles = await _context.Articles
+            .Where(article => article.UserId == userId)
+            .Select(article => _mapper.Map<ArticleDto>(article))
+            .ToListAsync();
+        
+        ThrowIfEmptyList(articles);
+        
```

Comment:
it is very small logic, so check the list in this one method

Reply:
also, you can check the list for null by continue your linq chain:
.ToListAsync() ?? throw new Exception();